source/fundamentals/typescript.txt

```diff
@@ -0,0 +1,97 @@
+==========
+TypeScript
+==========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecols
+
+Overview
+--------
+
+In this guide, you can learn about the **TypeScript** specific features and limitiations
+of the MongoDB Node.js driver. TypeScript is a strongly typed programming
+language that compiles to JavaScript.
+
+All TypeScript features of the driver are optional, and all valid JavaScript
+code written with the driver is also valid TypeScript code. 
+
+For more information on TypeScript, see 
+`the official TypeScript website <https://www.typescriptlang.org/>`__.
+
+Features
+--------
+
+If you use TypeScript, you can specify a type for some classes in the driver.
+All classes that accept a type paramater in the driver have the default type
+:node-api-4.0:`Document <interfaces/document.html>`.
+
+The following classes accept any type that extends the ``Document``
+interface:
+
+- :node-api-4.0:`Collection <classes/collection.html>`
+- :node-api-4.0:`ChangeStream <classes/changestream.html>`
+
+You can pass a type parameter that extends the ``Document`` interface like this:
+
+.. literalinclude:: /code-snippets/typescript/extend-document.ts
+   :language: typescript
+   :linenos:
+
+The following classes accept any type parameter:
+
+- :node-api-4.0:`FindCursor <classes/findcursor.html>`
+- :node-api-4.0:`AggregationCursor <classes/aggregationcursor.html>`
+
+You can find an example that shows how to specify a type for the ``FindCursor``
+class in our 
+:ref:`Find a Document Usage Example <node-driver-findone-usage-example-code-snippet>`.
+
+Limitations
+-----------
+
+.. _node-driver-typescript-limitations-dot-notation:
+
+The driver cannot infer the type of values with keys containing **dot
```

Comment:
Suggestion:" The driver cannot determine ..."

Reply:
I agree that in most cases determine would be a better word, but [type inference](https://www.typescriptlang.org/docs/handbook/type-inference.html) is a specific feature of the typescript language, so I think it is better to use "infer" here as it has a precise meaning. Open to further discussion if you disagree!